org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java

```diff
@@ -432,6 +434,15 @@ public void updateProjectList(IProgressMonitor monitor) {
       }).collect(toList());
   }
 
+  @Override
+  public Optional<ISonarLintProject> getBoundProject(String projectKey) {
```

Comment:
Not sure of the validity of this method. In Eclipse, it is very common to have multiple projects bound the the same SonarQube project. Should probably return a list.